### PR TITLE
Adapt IDL harness test for battery status to Promise-returning factory

### PR DESCRIPTION
**Pull request by dontcallmedom**

### CODE REVIEW

**Comment by tobie** [comment]
Reviewers for this pull request are: @w3c/battery-status-reviewers.

File: battery-status/battery-interface-idlharness.html [hunk 1]
```diff
@@ -37,15 +40,22 @@
...
 [].forEach.call(document.querySelectorAll('script[type=text\\/plain]'), function(node) {
-  // replace 'EventHandler' and 'unrestricted double' unrecognized by idlharness.js
-  idls = node.textContent.replace(/EventHandler/g, 'Function?').replace(/unrestricted double/g, 'double');
+  idls = node.textContent;
   idl_array[(node.className === 'untested') ? 'add_untested_idls' : 'add_idls'](idls);
 });
-idl_array.add_objects({Navigator: ['navigator'], BatteryManager: ['navigator.getBattery()']});
-idl_array.test();
+t.step(function() {
+navigator.getBattery().then(function(bm) {
```
Comment (zqzhang) [inline battery-status/battery-interface-idlharness.html]: This assumes that 'navigator.getBattery()' is supported. If it is not supported, the test may return a timeout or behave like hang. Better add an assertion before this line?

Please also make proper indent.

**Comment by zqzhang** [comment]
LGTM with nits.

File: battery-status/battery-interface-idlharness.html (hunk 1 above)
Comment (zqzhang) [inline battery-status/battery-interface-idlharness.html]: For example, `assert_idl_attribute(navigator, 'getBattery', 'navigator must have getBattery attribute');` in https://github.com/w3c/web-platform-tests/blob/master/battery-status/battery-interface.js#L14

**Pull request merged**